internal/engine/execution/procedure.go

```diff
@@ -172,18 +172,6 @@ func (p *preparedAction) call(scope *precompiles.ProcedureContext, global *Globa
 		return fmt.Errorf(`%w: procedure "%s" requires %d arguments, but %d were provided`, ErrIncorrectNumberOfArguments, p.name, len(p.parameters), len(inputs))
 	}
 
-	// if procedure does not have view tag, then it can mutate state
-	// this means that we must have a readwrite connection
-	if !p.view {
-		tx, ok := db.(sql.AccessModer)
-		if !ok {
-			return errors.New("DB does not provide access mode needed for mutative action")
-		}
-		if tx.AccessMode() != sql.ReadWrite {
```

Comment:
Just to confirm, this was redundant with the special instruction appended when preparing the action?